core/dbt/parser/schemas.py

```diff
@@ -1014,23 +1019,39 @@ def patch_constraints(self, node, constraints):
             node.constraints = [ModelLevelConstraint.from_dict(c) for c in constraints]
 
     def _validate_constraint_prerequisites(self, model_node: ModelNode):
+
+        column_warn_unsupported = [
+            constraint.warn_unsupported
+            for column in model_node.columns.values()
+            for constraint in column.constraints
+        ]
+        model_warn_unsupported = [
+            constraint.warn_unsupported for constraint in model_node.constraints
+        ]
+        warn_unsupported = column_warn_unsupported + model_warn_unsupported
+
+        # if any constraint has `warn_unsupported` as True then send the warning
+        if any(warn_unsupported) and model_node.config.materialized not in [
+            "table",
+            "incremental",
+        ]:
```

Comment:
`materialization_enforces_constraints` doesn't exist on 1.5.latest so have to hardcode this instead.